src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
+        final T element = x.item;
+        final MyLinkedList.Node<T> next = x.next;
+        final MyLinkedList.Node<T> prev = x.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            x.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            x.next = null;
+        }
+
+        x.item = null;
+        size--;
+        return element;
+    }
+
     @Override
     public void add(T value) {
+        final MyLinkedList.Node<T> l = last;
+        final MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(l, value, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndex(index);
+
+        if (index == size) {
+            add(value);
+        } else {
+            MyLinkedList.Node<T> succ = node(index);
+            final MyLinkedList.Node<T> pred = succ.prev;
+            final MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(pred, value, succ);
+            succ.prev = newNode;
+            if (pred == null) {
+                first = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        MyLinkedList.Node<T> x = node(index);
+        T oldVal = x.item;
+        x.item = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (MyLinkedList.Node<T> x = first; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (MyLinkedList.Node<T> x = first; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
-    public int size() {
-        return 0;
+    public boolean isEmpty() {
+        return size == 0;
     }
 
     @Override
-    public boolean isEmpty() {
-        return false;
+    public int size() {
+        return size;
+    }
+
+    private void checkIndex(int index) {
```

Comment:
Private methods and classes should be after public ones in your class. Move the `checkIndex` and `checkAddIndex` methods below the `node` method.